ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/internal/PropertySupplier.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.internal;
+
+import java.io.Serializable;
+import java.util.Map;
+
+/** Supply addititonal properties for catalog operations. This allows properties to be injected. */
+public interface PropertySupplier {
+
+  /** Return properties to be added to query operations. */
+  Map<String, Serializable> queryProperties();
```

Comment:
❗ There are many other types of operation properties like update, delete, resource, storage, and source info. It might be a good idea to make this more generic so we don't need to add a new method for each new implementation of `Operation`.

Reply:
One suggestion would be a `getProperties` method that takes the  class that implements `Operation` of interest as a lookup key.

Reply:
I went with a string parameter that indicates the type of operation.